diff --git a/src/components/Buttons/Button.js b/src/components/Buttons/Button.js
index a02ea04..d7b615f 100644
--- a/src/components/Buttons/Button.js
+++ b/src/components/Buttons/Button.js
@@ -1,16 +1,40 @@
-import React, { Component } from "react";
-
+import React, { Component } from 'react'
+import { getAllProperties, setProperty, deleteProperty } from '../../services/policyHandler'
 class Button extends Component {
-  
-  render() {
-    return (
-  <section><button type="button" className="btn btn-danger">Apply SSU Rollout Policy</button></section>
-
-  );
-}
+  onClick() {
+    console.log('onClick')
+    try {
+      let testProp = {
+        coverage: 1,
+        coverage_seed: 0,
+        filter: {
+          group: {
+            group1: '1',
+          },
+        },
+        key: 'Test::Enable',
+        ns: 'ada',
+        value: '51',
+      }
+      //getAllProperties()
+      setProperty(testProp)
+      //deleteProperty(testProp)
 
+      console.log('onClick done')
+    } catch (error) {
+      console.log('call failed')
+    }
+  }
 
+  render() {
+    return (
+      <section>
+        <button type='button' className='btn btn-danger' onClick={this.onClick}>
+          Apply SSU Rollout Policy
+        </button>
+      </section>
+    )
+  }
 }
 
-export default Button;
-
+export default Button